security.php

```diff
@@ -54,56 +55,148 @@ function wpcom_vip_is_restricted_username( $username ) {
 		|| WPCOM_VIP_MACHINE_USER_EMAIL === $username;
 }
 
+function _wpcom_vip_login_cache_keys( $raw_username ) {
+	$username = vip_strict_sanitize_username( $raw_username );
+
+	// phpcs:ignore WordPressVIPMinimum.Variables.ServerVariables.UserControlledHeaders
+	$ip = filter_var( $_SERVER['REMOTE_ADDR'] ?? '', FILTER_VALIDATE_IP, [ 'options' => [ 'default' => '' ] ] );
+
+	return [
+		'ip_username_cache_key' => $ip . '|' . $username,
+		'ip_cache_key'          => $ip,
```

Comment:
Yep

Reply:
Is an empty cache key legit? Is that checked for somewhere?

Reply:
I don't know. But seeing as it worked until this point (at least [years](https://github.com/Automattic/vip-go-mu-plugins/blame/develop/security.php#L70)) I think we are OK to keep it as is. Or do you think changing it is worth the increased risk of breaking things? If yes what do you suggest?

Reply:
<img width="559" alt="Screenshot 2023-06-13 at 11 47 18" src="https://github.com/Automattic/vip-go-mu-plugins/assets/88371/ce194943-95d0-4950-8e85-1359b5c767a0">

It seems that an empty cache key can't be used, so this may be "working" in the surrounding logic by accident instead of on purpose.

Reply:
Good, thanks for checking.